apps/explorer/src/hooks/useErc20.ts

```diff
@@ -113,12 +62,28 @@ export function useMultipleErc20(
 
   const [isLoading, setIsLoading] = useState(false)
   const [errors, setErrors] = useState<Errors>({})
+  const { isLoading: isTokenListLoading, data: tokenListTokens } = useTokenList(networkId)
 
   const erc20s = useMultipleErc20sState({ networkId, addresses })
   const saveErc20s = useSaveErc20s(networkId)
 
+  const fromTokenList = useMemo(
+    () =>
+      addresses.reduce((acc, address) => {
+        const token = tokenListTokens[address.toLowerCase()]
+        if (token) {
+          acc[address] = token
+        }
+        return acc
+      }, {}),
+    [addresses, tokenListTokens]
+  )
+
   // check what on globalState has not been fetched yet
-  const toFetch = useMemo(() => addresses.filter((address) => !erc20s[address]), [addresses, erc20s])
+  const toFetch = useMemo(
+    () => (isTokenListLoading ? [] : addresses.filter((address) => !erc20s[address] && !fromTokenList[address])),
```

Comment:
Only try to fetch if token list is not loading, to avoid triggering unnecessary onchain requests